pypesto/petab/importer.py

```diff
@@ -415,6 +437,36 @@ def create_objective(
             par_ids = [x for x in par_ids if x not in inner_parameter_ids]
 
             # FIXME: currently not supported with hierarchical
+            if 'guess_steadystate' in kwargs and kwargs['guess_steadystate']:
+                warnings.warn(
+                    "`guess_steadystate` not supported with hierarchical optimization. Disabling `guess_steadystate`."
+                )
+            kwargs['guess_steadystate'] = False
+
+        if self._ordinal:
+            inner_solver_options = kwargs.pop(
+                'inner_solver_options', self._inner_solver_options
+            )
+            inner_problem_method = inner_solver_options.get(
+                'method', self._inner_solver_options.get('method')
+            )
```

Comment:
Not sure if this is ideal, since a user might find it unintuitive that some of their options provided via `PetabImporter.__init__(..., inner_solver_options)` get used as default here, but not others. I would suggest either using all as default, or none, with a preference for none -- better if the user explicitly specifies every time they customize via `create_objective` -- otherwise, there are two sources of defaults: `PetabImporter._inner_solver_options` and `OptimalScalingInnerSolver.get_default_options`.

Reply:
Yes, that's true. But then again, I don't see another clean way of being able to specify the options through `PetabImporter.__init__(..., inner_solver_options)`.
Possibly it's just best to revert to without that possibility.

Reply:
Hm, just to elaborate, what I meant is, if you opt for no defaults, then, inside `create_objective`:
```python
inner_solver_options = (
    inner_solver_options
    if inner_solver_options is not None
    else self._inner_solver_options
)
```

---

If you opt for defaults, then, inside `create_objective`:
```python
inner_solver_options = {
    **self._inner_solver_options,
    **inner_solver_options,
}
```

---

At the moment, you have the equivalent of, inside `create_objective`:
```
inner_solver_options = {
    **{
        k:v
        for k, v in self._inner_solver_options.items()
        if k == "method"
    },
    **inner_solver_options,
}
```

Reply:
Thanks for the clarification. Yes, I had a weird thing in `create_objective` that if you provided `inner_solver_options=None` then it would take `inner_solver_options = None`, but the method would be taken from `self._inner_solver_options`.

What I was trying to do is the first option you wrote, not having in mind the possibility that the user can provide`inner_solver_options=None` in `create_objective`. 

But now that I see your second suggestion, that seems like the best option to use, to use what options are provided in `create_objective`, and add options provided in `PetabImporter.__init__(..., inner_solver_options)`. And then again do the same with the default inner_solver_options.

Do you think this is unnecessary? And confusing (having a 2-level-default options, user provided and internal from inner_solver)

Reply:
Hm, my preference is for the first case, but I'm fine with either. You could emit a warning/exception if `self._inner_solver_options` is not empty when the user supplies `inner_solver_options` to `create_objective`. In this way, these can be treated as separate, incompatible use cases.

As for a user-supplied `None` -- it should be made clear in the docs that a user supplies `{}` to indicate an empty `inner_solver_options`, not `None`. Then, you can treat `None` as the lack of any user input for this argument.

Reply:
Yes, on second thought, you're right. 
As a user, I would hate the interaction of pulling the additional settings from `PetabImporter.__init__(..., inner_solver_options)` even if I give something in objective. 
Cool, thank you for the discussion. 

Will make it nice :)